src/Engine/RulesComponent.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\RulesComponent.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\rules\Context\ContextDefinitionInterface;
+
+/**
+ * Handles executable Rules components.
+ */
+class RulesComponent {
+
+  /**
+   * The rules execution state.
+   *
+   * @var \Drupal\rules\Engine\RulesStateInterface
+   */
+  protected $state;
+
+  /**
+   * Definitions for the context used by the component.
+   *
+   * @var \Drupal\rules\Context\ContextDefinitionInterface[]
+   */
+  protected $contextDefinitions;
+
+  /**
+   * Definitions for the context provided by the component.
+   *
+   * @var \Drupal\rules\Context\ContextDefinitionInterface[]
+   */
+  protected $providedContextDefinitions;
+
+  /**
+   * The expression.
+   *
+   * @var \Drupal\rules\Engine\ExpressionInterface
+   */
+  protected $expression;
+
+  /**
+   * Constructs the object.
+   *
+   * @param \Drupal\rules\Engine\ExpressionInterface $expression
+   *   The expression of the component.
+   *
+   * @return static
```

Comment:
I looked it up - this is correct although our docs are not 100% accurate. opened https://www.drupal.org/node/2631930